js/pro/gate.js

```diff
@@ -741,23 +741,49 @@ module.exports = class gate extends gateRest {
     }
 
     async watchOrders (symbol = undefined, since = undefined, limit = undefined, params = {}) {
+        /**
+         * @method
+         * @name gate#watchOrders
+         * @description watches information on multiple orders made by the user
+         * @see https://www.gate.io/docs/developers/apiv4/ws/en/#orders-channel
+         * @see https://www.gate.io/docs/developers/futures/ws/en/#orders-api
+         * @see https://www.gate.io/docs/developers/options/ws/en/#orders-channel
+         * @param {string|undefined} symbol unified market symbol of the market orders were made in
+         * @param {int|undefined} since the earliest time in ms to fetch orders for
+         * @param {int|undefined} limit the maximum number of  orde structures to retrieve
+         * @param {object} params extra parameters specific to the gate api endpoint
+         * @param {string} params.type spot, margin, swap, future, or option. Required if listening to all symbols.
+         * @param {boolean} params.isInverse if future, listen to inverse or linear contracts
+         * @returns {[object]} a list of [order structures]{@link https://docs.ccxt.com/en/latest/manual.html#order-structure
+         */
+        let type = undefined;
+        let market = undefined;
         if (symbol === undefined) {
-            throw new ArgumentsRequired (this.id + ' watchOrders requires a symbol argument');
-        }
-        await this.loadMarkets ();
-        const market = this.market (symbol);
-        symbol = market['symbol'];
-        let type = 'spot';
-        if (market['future'] || market['swap']) {
-            type = 'futures';
-        } else if (market['option']) {
-            type = 'options';
+            const [ defaultType ] = this.handleMarketTypeAndParams ('watchOrders', undefined, params);
+            type = defaultType;
+        } else {
+            await this.loadMarkets (false, params);
```

Comment:
@pcriadoperez we should always load the markets, because it will be needed to parse the orders regardless of whether the user provides a symbol or not